src/opts.rs

```diff
@@ -141,11 +141,13 @@ pub struct Opts {
 impl Opts {
     pub fn process(&mut self) {
         LogLevel::from_verbosity_flag_count(self.verbose).apply();
-
+        let me = self.clone();
+        let mut data_dir = self.data_dir.to_string_lossy().to_string();
+        me.process_dir(&mut data_dir);
+        self.data_dir = PathBuf::from(data_dir);
```

Comment:
I think we can go for

```suggestion
        let mut me = self.clone();
        me.data_dir = shellexpand::tilde(me.data_dir).to_string();
```

No need to use `process_dir` which also does `{data_dir}` substitution and opens attack vector with the call stack overflow

Reply:
I don't know why I stuck with using `process_dir`, like that it's definitely better.

Reply:
I needed to do some type conversions (in e4f7d13) to get it compile.